src/library/scala/collection/LinearSeq.scala

```diff
@@ -34,9 +34,22 @@ object LinearSeq extends SeqFactory.Delegate[LinearSeq](immutable.LinearSeq)
 /** Base trait for linear Seq operations */
 trait LinearSeqOps[+A, +CC[X] <: LinearSeq[X], +C <: LinearSeq[A] with LinearSeqOps[A, CC, C]] extends Any with SeqOps[A, CC, C] {
 
-  // To be overridden in implementations:
-  def isEmpty: Boolean
+  /** @inheritdoc
+   *
+   *  Note: *Must* be overridden in subclasses. The default implementation returns true if `knownSize` is not positive.
+   */
+  override def isEmpty: Boolean //= knownSize <= 0
```

Comment:
I don’t understand the “default” implementation. It means that `isEmpty` would return `true` if the size of the collection cannot be computed cheaply (ie, if `knownSize` is negative).

Reply:
There is no good default. It must be overridden.